algorithms/kernel/covariance/oneapi/cl_kernels/covariance_kernels.cl

```diff
@@ -86,8 +88,11 @@ DECLARE_SOURCE(
             }
             else if (global_row_id != global_col_id)
             {
-                algorithmFPType sqrtRowElement = (algorithmFPType)(1.0) / sqrt(crossProductRowElement);
-                algorithmFPType sqrtColElement = (algorithmFPType)(1.0) / sqrt(crossProductColElement);
+                //Handling cases when inversion can not be performed.
+                algorithmFPType sqrtRowElement =
```

Comment:
This not affect performance? May be use `max`/`min` function?

Reply:
I'll run benchmarks to understand if it affect performance.